apptainer/moose-dev.def

```diff
@@ -71,23 +77,16 @@ From: {{ APPTAINER_FROM }}
     # Create list of Conda Packages to install
     MAMBA_PACKAGES="moose-tools=${MOOSE_TOOLS_VERSION}"
     # Install packages
-    mamba create -yq -n ${MOOSE_MAMBA_ENV} python=${PYTHON_VERSION} ${MAMBA_PACKAGES}
+    mamba create -yq -n ${MOOSE_CONDA_ENV_NAME} python=${PYTHON_VERSION} ${MAMBA_PACKAGES} openssl=3.1.4
```

Comment:
pesky openssl.

I am a little confused why I need to pin this while we build Conda packages for Mac targets. But we need this here for runtime Linux targets =D

Reply:
I don't think we need this anymore... but we're in too deep and can remove it on the next update

Reply:
Went ahead and removed this